### DISCUSSION TITLE
[Question] APK long packaging time

### DISCUSSION THREAD

**Discussion by Maskedviper** [flet-dev/flet#2625] [Q&A]
### Question

We are working on creating an app using Python and the Flet framework, after following several videos and links we are still running into an error in packaging the code as an apk file using the Flet framework. When we add the libraries one at a time they modulenotfounderror Is removed for the previous library not included but when we include all libraries in one go its taking hours for the packaging process and it still doesn't end up packaging. So I was wondering is this normal? Or how long does it take to package usually?

![WhatsApp Image 2024-02-12 at 12 12 55 PM](https://github.com/flet-dev/flet/assets/10101674/961843d1-6235-42f7-b5a6-3e3fabd465da)

### ------------------------------------------------------

- [X] I have searched for answers to my question both in the issues and in previous discussions.

**Comment by 09u2h4n** [comment]
Can you send the log `flet build apk app_name/ -vv`

Mine is not that long (I didn't measure time) but sometimes i need to abort then restart the command

**Comment by kalatawei** [comment]
**i have the same problem ! still not solved.**

**Reply by 09u2h4n** [reply]
Maybe your flet version is not the latest one try `pip install -U flet` to update to latest version.

**Reply by kalatawei** [reply]
(flet_env) C:\Users\pc\Desktop\FletProject\flet_test>pip show flet
Name: flet
Version: 0.20.2
Summary: Flet for Python - easily build interactive multi-platform apps in Python
Home-page:
Author: Appveyor Systems Inc.
Author-email: hello@flet.dev
License: Apache-2.0
Location: C:\Users\pc\Desktop\FletProject\flet_env\Lib\site-packages
Requires: cookiecutter, flet-runtime, packaging, qrcode, watchdog, websocket-client, websockets
Required-by:

**i'm using the latest version !**

_**did you try to build with any of libraries mentioned in the image above ?**_

**Comment by tanmay-bhatgare** [comment]
You should check Google IDX code editor for packaging flet app, mine take too long like 30 to 40 minutes in VS code(still causes an error) but when running on IDX it just take approx. 10 minutes. You just have to make a flutter project and then delete all the flutter related files not _**dev**_ folder. and configure some variables like adding python 10 package and pip too, and then try to run `flet build apk` or any other platform command

**Reply by ndonkoHenri** [reply]
@tanmay-bhatgare, can you please better explain how to have Flet running there? I gave it a try, but was unsuccessful.

**Reply by tanmay-bhatgare** [reply]
Hello @ndonkoHenri,

To package your application, you should start by creating a Flutter project if you intend to build a mobile app. After that, remove all files related to Flutter, as they are not required for your Python application.

For the installation of Python, you should add `pkgs.python10Full` and `pkgs.python310Packages.pip` to the list of packages in the `dev.nix` file. Once this is done, rebuild your environment.

Next, create a virtual environment and proceed to install Flet. Please note that you can only use `flet run --web` since IDX does not support a built-in GUI. However, you can still use this method to package your application into an `.apk` file or any other format you desire.

I hope this helps!

**Reply by ndonkoHenri** [reply]
> I hope this helps!

Hat to tweak it a little, but it worked, thanks!
Video tutorial: https://youtu.be/TiW-JTd4EP0
